### ISSUE TITLE
Microsoft C/C++ compiler and linker options are not covered by the hardening guide

### ISSUE THREAD

**Issue by balteravishay** [ossf/wg-best-practices-os-developers#150]
The current version of the C/C++ hardening guide is incredibly helpful in supporting  GCC and CLang.
Opening this issue to track the work that our team with @michaelcfanning and @GabrielDosReis intends to contribute towards adding support for the Microsoft C/C++ compiler and linker options to the hardening guide.

**Comment by SecurityCRob** [comment]
Have we been able to get MSFT to engage on this?    @gkunz @thomasnyman @david-a-wheeler

**Comment by thomasnyman** [comment]
@balteravishay, @GabrielDosReis: any update on the Microsoft C/C++ compiler and linker options or should we close this issue?

**Comment by michaelcfanning** [comment]
Yes, we reviewed a preliminary initial draft which I pushed to a public repository as one of my final acts before taking the MS VRP. :)

I'm no longer working at Microsoft but still active in open-source development and would be glad to help bring this content into OSSF (along with @GabrielDosReis of course). Any suggestions on next steps?

https://github.com/microsoft/msvc-compiler-hardening-guide

**Comment by gkunz** [comment]
Thank you very much, @michaelcfanning and @GabrielDosReis!

We noticed that that repo does not yet have a LICENSE file. Would be great if you could add one.

Also adding @scovetta for visibility as suggested on today's OpenSSF TAC call.
